src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputUnderscoreUsedInNames.java

```diff
@@ -5,112 +5,119 @@
 
   class InnerBad {
-    String guava_33_4_7 = ""; // violation due to '_' between digit and letter
-    String guava33_4_7 = "";
+    int guava_33_4_7; // violation, due to _ between digit and letter, 'must match pattern'
+    int guava33_4_7;
 
-    String guava33_4_8_ = ""; // violation, underscore at the end
-    String guava33_4_8 = "";
+    int guava33_4_8_; // violation, _ at the end, 'must match pattern'
+    int guava33_4_8;
 
-    String jdk_8_90 = ""; // violation due to '_' between digit and letter
-    String jdk8_90 = "";
+    int jdk_8_90; // violation, due to _ between digit and letter, 'must match pattern'
+    int jdk8_90;
 
-    String jdk8_91_ = ""; // violation, underscore at the end
-    String jdk8_91 = "";
+    int jdk8_91_; // violation, _ at the end, 'must match pattern'
+    int jdk8_91;
 
-    String kotlin_1_9_24 = ""; // violation due to '_' between digit and letter
-    String kotlin1_9_24 = "";
+    int kotlin_1_9_24; // violation, due to _ between digit and letter, 'must match pattern'
+    int kotlin1_9_24;
 
-    // violation below, '_' not allowed between lowercase character sequences
-    String kotlin_version1_9_24 = "";
+    // violation below, _ not allowed between lowercase character sequences, 'pattern'
+    int kotlin_version1_9_24;
 
-    String kotlinVersion1_9_24 = "";
+    int kotlinVersion1_9_24; // violation, false-positive, 'must match pattern'
 
-    String kotlin1_9_25_ = ""; // violation, underscore at the end
-    String kotlin1_9_25 = "";
+    int kotlin1_9_25_; // violation, _ at the end, 'must match pattern'
+    int kotlin1_9_25;
   }
 
   static class MultipartVersioningNamesBad {
 
-    void guava_34_4_6() {} // violation due to '_' between digit and letter
+    void guava_34_4_6() {} // violation, due to _ between digit and letter, 'must match pattern'
 
     void guava34_4_6() {}
 
-    void kotlin_2_9_94() {} // violation due to '_' between digit and letter
+    void kotlin_2_9_94() {} // violation, due to _ between digit and letter, 'must match pattern'
 
     void kotlin2_9_94() {}
 
-    void gradle_9_5_1() {} // violation due to '_' between digit and letter
+    void gradle_9_5_1() {} // violation, due to _ between digit and letter, 'must match pattern'
 
     void gradle9_5_1() {}
 
-    void jdk_9_0_392() {} // violation due to '_' between digit and letter
+    void jdk_9_0_392() {} // violation, due to _ between digit and letter, 'must match pattern'
 
     void jdk9_0_392() {}
 
-    // violation below, '_' not allowed between lowercase character sequences
+    // violation below, _ not allowed between lowercase character sequences, 'pattern'
     void kotlin_lang1_9_2() {}
 
-    void kotlinLang1_9_2() {}
+    void kotlinLang1_9_2() {} // violation, false-positive, 'must match pattern'
 
-    // violation below, '_' not allowed between lowercase character sequences
+    void convertToKotlinVersion1_9_24() {} // violation, false-positive, 'must match pattern'
```

Comment:
Ok, we need some suppression to avoid false positives.
Suppression config and resulted false negatives should be marked until bsime ticket, please create it, 
And we will merge.